src/colmap/estimators/bundle_adjustment.h

```diff
@@ -64,19 +64,27 @@ struct BundleAdjustmentOptions {
   // Whether to print a final summary.
   bool print_summary = true;
 
-  // Whether to use Ceres' CUDA sparse linear algebra library, if available.
+  // Whether to use Ceres' CUDA linear algebra library, if available.
   bool use_gpu = false;
   std::string gpu_index = "-1";
 
-  // Minimum number of residuals to enable multi-threading. Note that
-  // single-threaded is typically better for small bundle adjustment problems
-  // due to the overhead of threading.
-  int min_num_residuals_for_multi_threading = 50000;
+  // Heuristic threshold to switch from CPU to GPU based solvers.
+  // Typically, the GPU is faster for large problems but the overhead of
+  // transferring memory from the CPU to the GPU leads to better CPU performance
+  // for small problems. This depends on the specific problem and hardware.
+  int min_num_images_gpu_solver = 50;
+
+  // Heuristic threshold on the minimum number of residuals to enable
+  // multi-threading. Note that single-threaded is typically better for small
+  // bundle adjustment problems due to the overhead of threading.
+  int min_num_residuals_for_cpu_multi_threading = 50000;
 
   // Heuristic thresholds to switch between direct, sparse, and iterative
   // solvers. These thresholds may not be optimal for all types of problems.
-  int max_num_images_direct_dense_solver = 50;
-  int max_num_images_direct_sparse_solver = 1000;
+  int max_num_images_direct_dense_cpu_solver = 50;
```

Comment:
Do we want to pybind these attributes as well?

Reply:
Thanks, done